internal/store/engine_state.go

```diff
@@ -65,7 +65,8 @@ type EngineState struct {
 	TiltfileState ManifestState
 
 	// from GitHub
-	LatestTiltBuild model.TiltBuild
+	LatestTiltBuild     model.TiltBuild
+	CheckVersionUpdates bool
```

Comment:
can we put the whole VersionSettings in the engine state, for consistency?

want to move towards a world where we have one Tiltfile result data Model on the engine state, rather than the current world where there are 3 or 4 artisanal data types in-between that are hard to reason about

Reply:
@nicks would this mean that the struct is duplicated between `internal/tiltfile/analytics` and `internal/model` ?

Reply:
Or should `internal/tiltfile/analytics` point to `internal/model`?

Reply:
take a look at how DockerPruneSettings does it